sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -80,7 +98,7 @@ private[spark] object CosmosClientCache extends BasicLoggingTrait {
             // retrieved it form the cache before we remove it here
             // so if it is actively used now we need to keep a reference and close it
             // when it isn't used anymore
-            if (existingClientCacheMetadata.refCount.get() == 0) {
+            if (forceClosure || existingClientCacheMetadata.refCount.get() == 0) {
```

Comment:
Does the `existingClientCacheMetadata.refCount.get()` is not zero when force closing?
I am wondering if we have this check already, why we can't leverage it?

Reply:
Also, if we are closing this client, should we update the `refCount` in case of forceClosures, or it doesn't matter since we are force closing now?

Reply:
> Does the `existingClientCacheMetadata.refCount.get()` is not zero when force closing? I am wondering if we have this check already, why we can't leverage it?

refcount is irrelevant - we only force closure on shutdown pass. At which point refCount isn't relevant anymore.